### PR TITLE
Icon convert to typescript

### PR DESCRIPTION
**Pull request by danethurber**
let's try this again

### CODE REVIEW

**Comment by jaketrent** [comment]
All checks pass.  Well, I'm sold.  :)  Reviewing...

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ default:
 
 before_script:
   - yarn config set ignore-engines true
+  - yarn config set workspaces-experimental true
```
Comment (jaketrent) [inline .gitlab-ci.yml:31]: What is this voodoo?

File: packages/icon/cli/clean-svgs.js [hunk 2]
```diff
@@ -2,17 +2,25 @@ const { promises: fs } = require('fs')
...
     { removeXMLNS: true },
     { removeDimensions: true },
     { removeAttrs: { attrs: '(stroke|fill)' } }
   ]
 })
-const regex = /aria-label=(?:"|')([\w ]+)(?:"|')/
-const addAria = (svg, name) =>
-  regex.test(svg) ? svg : svg.replace(/<svg/, `<svg aria-label="${name} icon"`)
+
+const ariaLabelRegEx = /aria-label=(?:"|')([\w ]+)(?:"|')/
```
Comment (jaketrent) [inline packages/icon/cli/clean-svgs.js:14]: I don't remember this change last time.  But this time, I'm not seeing any change that seems significant.  Am I missing something?

File: packages/icon/cli/generate-components.js [hunk 3]
```diff
@@ -5,34 +5,85 @@ const SVGO = require('svgo')
...
-    ${svg.replace(
-      regex,
-      "role='img' {...(ariaLabel && { 'aria-label': ariaLabel })}>"
-    )}
-  </Icon>
-)})
-${componentName}.displayName = '${componentName}'
-${componentName}.propTypes = {
-  'aria-label': PropTypes.string,
+exports.generateComponents = async ({
```
Comment (jaketrent) [inline packages/icon/cli/generate-components.js:8]: Is this just a file resort?  Or are there some significant changes you would highlight here?

File: packages/icon/tsconfig.build.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    ".storybook",
+    "node_modules",
+    "dist",
+    "**/*.spec.*",
+    "**/*.story.*"
+  ],
+  "compilerOptions": {
+    "declaration": true,
+    "outDir": "dist",
+    "rootDir": "src"
```
Comment (jaketrent) [inline packages/icon/tsconfig.build.json:14]: strict eventually?

**Review by jaketrent — APPROVED** [review]
Looks good locally.  Let the CI rip!

File: .gitlab-ci.yml (hunk 1 above)
Comment (danethurber) [reply .gitlab-ci.yml:31]: this actually doesn't fix the issue. workspaces is not enabled by default. the real issue was a circular dep

File: packages/icon/cli/clean-svgs.js (hunk 2 above)
Comment (danethurber) [reply packages/icon/cli/clean-svgs.js:14]: nothing significant. just trying to name the vars

File: packages/icon/cli/generate-components.js (hunk 3 above)
Comment (danethurber) [reply packages/icon/cli/generate-components.js:8]: the only significant thing was the new component code. everything else was just trying to make sure i left it better and more understandable

File: packages/icon/tsconfig.build.json (hunk 4 above)
Comment (danethurber) [reply packages/icon/tsconfig.build.json:14]: yeah. i'll follow up

Comment (danethurber) [reply packages/icon/tsconfig.build.json:14]: i checked and there isn't anything in the src. a couple things in stories we can update

**Pull request merged**